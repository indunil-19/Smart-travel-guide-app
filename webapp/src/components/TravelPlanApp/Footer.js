import { Flex, HStack, VStack,Text ,Heading,Divider} from "@chakra-ui/layout"
import { Link } from "react-router-dom"

export const Footer=()=>{
    return(
        <>
        <Flex width="full" bg="black"  mt="165px" color="white" justifyContent="space-around" p={10}  >
             <VStack>
                <HStack spacing={10}>
                <VStack alignItems="start" spacing={8}>

                     
                     
                     <Heading as="h3" size="lg">
                        About Us
                    </Heading>
                 
                    

                    <Heading as="h3" size="lg">
                        Contact Us
                    </Heading>
                  
                </VStack>

                <Divider orientation="vertical"   variant="solid" p={3}  />

                <VStack>
                     
                     <>

                     <iframe src="https://www.facebook.com/plugins/page.php?href=https%3A%2F%2Fwww.facebook.com%2FWonderDiscoveryTravelSriLanka%2F&tabs=messages&width=340&height=500&small_header=false&adapt_container_width=true&hide_cover=false&show_facepile=true&appId" width="340" height="500" style={{border:"none",overflow:"hidden"}} scrolling="no" frameborder="0" allowfullscreen="true" allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"></iframe>                     </>
                </VStack>
                </HStack>
                       <Divider  orientation="horizontal"   variant="solid" p={3}  />
                </VStack>
               
        </Flex>
       


        </>
    )
}