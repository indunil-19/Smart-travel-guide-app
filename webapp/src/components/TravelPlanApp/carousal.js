import React, { useEffect,useState } from "react";
// import  "react-responsive-carousel/lib/styles/carousel.min.css"; 
import { Carousel } from 'react-responsive-carousel';




 const Carousal=({photos=[]})=>{
    const [references, setReferences]=useState([])
    useEffect(()=>{
            setReferences(photos)
    }, [photos])
    return(
        <Carousel>
                
                {
                    references.map((reference)=>{
                        return(
                            <div >
                            <img data-testid="image"  src={`https://maps.googleapis.com/maps/api/place/photo?maxwidth=800&photo_reference=${reference.photo_reference}&key=AIzaSyChMTwAb_hWwYdvcM_gSGcx84k_al-EtIA`} 
                             />
                            
                        </div>
                        )
                    })
                }
                
            </Carousel>
    )
}
export default Carousal