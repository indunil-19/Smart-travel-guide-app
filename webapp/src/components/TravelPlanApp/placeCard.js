import React , {useEffect,useState}from "react"
import { useHistory } from "react-router"
import { Box, Flex ,HStack, VStack} from "@chakra-ui/layout"
import { Image,Button,Badge, Avatar , Heading} from "@chakra-ui/react"
import { StarIcon } from "@chakra-ui/icons"
import StarRatings from "react-star-ratings"

export const PlaceCard=({photo="",index,name,address, types=[], rating, place_id})=>{
    const [link,setLink]=useState("")
    const history=useHistory()
    useEffect(() => {
        
        
        setLink(photo)
    }, [photo])
    return(
        <Flex flexDirection="row" margin={5} borderWidth="1px" borderRadius="lg" p={5} width="100%">
                           <Image src={link}  width="400px" height="300px" objectFit="cover" borderRadius="lg"  />
                           <VStack spacing={3.5} alignItems="start" p={5}>
                                 <Box bg="green.200" p={3} borderRadius="full" border="green" borderWidth="1px" ><Heading as="h4" size="md">{index.toString()}</Heading></Box>
                                 <Box lineHeight="tall" fontWeight="semibold" as="h4"> {name}</Box>
                                 <Box color="gray.600" fontSize="sm">{address}</Box>
                                <HStack>
                                        <>    {types.map((item)=>{
                                                return(
                                                    <Badge borderRadius="full" px="2" colorScheme="teal">{item}</Badge>
                                                )
                                            })}
                                        </> 
                                         
                                </HStack>

                                <HStack>
                                        {/* <StarIcon color={"teal.500"} />
                                        <StarIcon color={"teal.500"} />
                                        <StarIcon color={"teal.500"} />
                                        <StarIcon color={"teal.500"} />
                                        <StarIcon color={"teal.500"} /> */}


                                        <StarRatings
                                        rating={rating}
                                        starRatedColor="teal"
                                        numberOfStars={5}
                                        name='rating'
                                        starDimension="20px"
                                        starSpacing="3px"
                                        />
                                        <Box>rating {rating}</Box>
                                        
                                </HStack>
                               
                                        <Button colorScheme="teal" variant="outline" onClick={()=>{
                                            history.push('/travelPlan/viewpoi/'+place_id)
                                        }}>
                                            See more...
                                        </Button>

                           </VStack>

        </Flex>
    )
}