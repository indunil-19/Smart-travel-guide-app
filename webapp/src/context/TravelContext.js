import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 