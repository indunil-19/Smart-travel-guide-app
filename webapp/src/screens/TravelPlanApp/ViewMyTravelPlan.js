import { TravelPlan } from "./TravelPlan"
import { Button } from "@chakra-ui/button"
import { VStack } from "@chakra-ui/layout"
import { AiFillStar } from "react-icons/ai";
import { useHistory } from "react-router"
import React from "react";

export  const ViewMyTravelPlan=()=>{
    const history=useHistory()

    return(
        <>
       <TravelPlan />
        <VStack position="fixed" bottom="105px" right="0" p={3} >

                <Button colorScheme="pink" size="lg" borderRadius="100%" data-testid="submit" onClick={()=>{
                    history.push('/travelPlan/review')
                }}>
                    <AiFillStar />                 
                </Button>
        </VStack>
        </>
    )
}