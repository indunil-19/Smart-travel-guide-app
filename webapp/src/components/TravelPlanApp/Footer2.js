import { Flex ,Text } from "@chakra-ui/layout"
import React from "react"
import Footer from "./Footer"

 const Footer2=()=>{
    return(
        <>
        <Flex width="full" bg="black" color="white"  p={5} flexDirection="column" alignItems="center" >

              <Text fontSize="xs">@university of moratuwa 2021/09/27</Text>


        </Flex>
        </>
    )
}
export default Footer2