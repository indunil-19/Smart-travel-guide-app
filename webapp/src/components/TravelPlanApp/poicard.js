import { Divider, Flex, HStack, Text,  } from "@chakra-ui/layout"
import React from "react"
import { Heading } from "@chakra-ui/layout";
import { IoLocation } from "react-icons/io5";
import { Tag } from "@chakra-ui/tag";
import { Carousal } from "../../components/TravelPlanApp/carousal"
import { Review } from "../../components/TravelPlanApp/review"


 const PoiCard=({data={}})=>{
    return(
        <Flex  flexDirection="column" alignItems="center" width="70%" marginX="auto" boxShadow="dark-lg" padding={4} marginTop={2}>            
            

            <HStack><IoLocation/>
            <Heading padding={5}>{data.name}</Heading>
            </HStack>

            <Carousal photos={data.photos} />

            <HStack spacing={6}  p={2}> 
                {(data.types)  && data.types.map((type)=>{
                    return(
                        <Tag  variant="solid" colorScheme="teal">
                            {type}
                        </Tag>
                    )
                })}
            </HStack>

            <Text fontSize="md">{data.formatted_address}</Text>
            
         




            <Divider  padding="5" />
            
            <Text fontSize="3xl">User reviews</Text>
            
            <>
            {(data.reviews) && data.reviews.map((review)=>{
                return(
                    <Review  author_name={review.author_name} profile_photo_url={review.profile_photo_url} relative_time_description={review.relative_time_description} text={review.text} rating={review.rating} author_url={review.author_url}/>
                )
            })}
            
            
            </>
                       
        </Flex>
    )
}
export default PoiCard