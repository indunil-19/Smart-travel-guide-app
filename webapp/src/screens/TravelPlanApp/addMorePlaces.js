import { useContext, useState, useEffect } from "react"
import { findPois, addPoiToPlan } from "../../services/EditPlanServices"
import { useParams } from 'react-router'
import { TravelContext } from "../../context/TravelContext"
import { PlaceCard } from "../../components/TravelPlanApp/placeCard"
import { Flex, HStack ,VStack} from "@chakra-ui/layout"
import { GrAddCircle } from "react-icons/gr";
import { Button } from "@chakra-ui/button"
import { useHistory } from "react-router"
import { Tabs, TabList, TabPanels, Tab, TabPanel,Alert} from "@chakra-ui/react"
import { AlertIcon } from "@chakra-ui/alert"
import Autocomplete from "react-google-autocomplete";

export const AddMorePlaces=()=>{
    const history=useHistory()
    const {day}=useParams()
    const [pois,setPois]=useState([])
    const [route,setRoute]=useState([])
    const {state, dispatch}=useContext(TravelContext)
    const [place1,setPlace1]=useState({})


    const [poisCustom,setPoisCustom]=useState([])
    const [routeCustom,setRouteCustom]=useState([])


    useEffect(() => {
        findPois(day, state.travelPlan, state.allpois).then((res)=>{
            console.log(res)
            setPois(res[0])
            setRoute(res[1])
        })
    }, [state])


    const addPoi=(day,poi, route)=>{
            addPoiToPlan(day,poi, route,state.travelPlan).then((res)=>{
                // console.log(res)
                dispatch({type:"set_travelPlan" , payload:{travelPlan:res}})
            })
    }

    return(
        <>
        <Tabs isFitted variant="line"  >
        <TabList  position="fixed" width="full" zIndex="8" bg="white" p={2} >
            <Tab >Place sugests for you</Tab>
            <Tab>Find by your self</Tab>
        </TabList>


        <TabPanels >


        <TabPanel>
        <Flex flexDirection="column" alignItems="center">
        {
            pois.length==0 ? 
            <>
                <Alert status="warning" mt="100px">
                    <AlertIcon />
                    Seems there is no pois for suggests. 
                </Alert>
            
            </> :<></>
        }
        
        { 
         pois.map((Item, index)=>{
            return(
                <>
                <HStack m={3}>

                <Button colorScheme="black" variant="outline" onClick={()=>{
                                addPoi(day,Item, route[index]);
                                history.push("/travelPlan/editPlan")

                 }}>
                     <GrAddCircle />           
                </Button>



                <PlaceCard  index={index+1} name={Item.name} address={Item.vicinity?Item.vicinity : Item.formatted_address  } photo={Item.photos ? Item.photos[0].photo_reference : "" } rating={Item.rating} place_id={Item.place_id}/>
                </HStack>
                </>
            )
        })

    }



        
        
        </Flex>
        </TabPanel>


        <TabPanel>


        <Flex flexDirection="column" alignItems="center" p={10}>
                <Autocomplete style={{width:"40%" ,height:"30px", padding:"5px", margin:"15px",background:"grey", borderRadius:"5px", color:"white"}}
                apiKey={"AIzaSyChMTwAb_hWwYdvcM_gSGcx84k_al-EtIA"}

                onPlaceSelected={(place) => {
                if(place.geometry.location){
                    setPlace1(place)
                    findPois(day, state.travelPlan, [place]).then((res)=>{
                        // console.log(res)
                        setPoisCustom(res[0])
                        setRouteCustom(res[1])
                    })
                }
                }}

                options={{
                    types: ['geocode' , 'establishment'],
                    componentRestrictions: { country: "LK" },
                    fields:["ALL"]
                }}
                >
                </Autocomplete>


        </Flex>






        </TabPanel>



        </TabPanels>
        </Tabs>
        </>
    )
}