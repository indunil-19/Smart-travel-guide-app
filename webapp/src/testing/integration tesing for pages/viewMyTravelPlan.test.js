import React from "react"
import {render,screen, cleanup,waitFor,fireEvent} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import 'whatwg-fetch'
import { ViewMyTravelPlan } from "../../screens/TravelPlanApp/ViewMyTravelPlan";
import { createMemoryHistory } from 'history'

describe('viewPlan page test', () => {
    beforeEach(()=>{      
    })
    
    afterEach(cleanup)

    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><ViewMyTravelPlan/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><ViewMyTravelPlan/></TravelProvider></MemoryRouter>,div)
    })
    it("should redirect to review page  if  button click",async()=>{
        const history = createMemoryHistory();
        render(<MemoryRouter><Router history={history}><TravelProvider><ViewMyTravelPlan/></TravelProvider></Router></MemoryRouter>)        
        
        fireEvent.click(screen.getByTestId("submit"))
        await waitFor(()=>{
            
            expect(history.location.pathname).toBe('/travelPlan/review')
        })
        
    })

})
