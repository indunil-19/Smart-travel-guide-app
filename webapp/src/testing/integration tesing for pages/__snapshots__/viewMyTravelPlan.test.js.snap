// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewPlan page test should match snapshot 1`] = `
Array [
  <div
    className="chakra-tabs css-0"
  >
    <div
      className="css-14mk567"
    >
      <img
        className="chakra-image__placeholder css-actnqx"
      />
      <h2
        className="chakra-heading css-17wl8jd"
      >
        0
         Day Trip in 
      </h2>
      <div
        className="chakra-stack css-1qwdsol"
      >
        <div
          aria-orientation="horizontal"
          className="chakra-tabs__tablist css-1xc77o7"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls="tabs-2815631040-2--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-5wl5ud"
            id="tabs-2815631040-2--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Travel Plan
          </button>
          <button
            aria-controls="tabs-2815631040-2--tabpanel--1"
            aria-selected={false}
            className="chakra-tabs__tab css-5wl5ud"
            id="tabs-2815631040-2--tab--1"
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Travel Route
          </button>
        </div>
      </div>
    </div>
    <div
      className="chakra-tabs__tab-panels css-8atqhb"
    >
      <div
        aria-labelledby="tabs-2815631040-2--tab-0"
        className="chakra-tabs__tab-panel css-sjt5nk"
        hidden={false}
        id="tabs-2815631040-2--tabpanel-0"
        role="tabpanel"
        tabIndex={0}
      >
        <div
          className="css-dvxtzn"
          data-testid="travelPlan"
        >
          <div
            className="chakra-stack css-84zodg"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 32C167.67 32 96 96.51 96 176c0 128 160 304 160 304s160-176 160-304c0-79.49-71.67-144-160-144zm0 224a64 64 0 1164-64 64.07 64.07 0 01-64 64z"
              />
            </svg>
            <span
              className="chakra-badge css-ifsvay"
            >
              9.00 a.m
            </span>
            <p
              className="chakra-text css-1tapsqs"
            >
              Start from colombo
            </p>
          </div>
          <div
            className="chakra-stack css-84zodg"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 32C167.67 32 96 96.51 96 176c0 128 160 304 160 304s160-176 160-304c0-79.49-71.67-144-160-144zm0 224a64 64 0 1164-64 64.07 64.07 0 01-64 64z"
              />
            </svg>
            <span
              className="chakra-badge css-ifsvay"
            >
              6.00 p.m
            </span>
            <p
              className="chakra-text css-1tapsqs"
            >
              End - 
              
            </p>
          </div>
        </div>
        <div
          className="chakra-stack css-1obfgbw"
        >
          <button
            className="chakra-button css-xdqcq1"
            data-testid="edit"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
              />
              <path
                d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
              />
            </svg>
          </button>
          <button
            className="chakra-button css-xdqcq1"
            data-testid="save"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11l5 5v11a2 2 0 0 1-2 2z"
              />
              <polyline
                points="17 21 17 13 7 13 7 21"
              />
              <polyline
                points="7 3 7 8 15 8"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        aria-labelledby="tabs-2815631040-2--tab-1"
        className="chakra-tabs__tab-panel css-sjt5nk"
        data-testid="maptab"
        hidden={true}
        id="tabs-2815631040-2--tabpanel-1"
        role="tabpanel"
        tabIndex={0}
      >
        <div
          className="css-1j2sprm"
        >
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="chakra-stack css-1xrxb8h"
  >
    <button
      className="chakra-button css-1fozia0"
      data-testid="submit"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
        />
      </svg>
    </button>
  </div>,
]
`;
